control/summary_table.go

```diff
@@ -71,27 +73,104 @@ func (st *SummaryTable) Fill(openControlData opencontrols.Data) (err error) {
 	return
 }
 
+// diffControlOrigination computes the diff of the control origination.
+func (st *SummaryTable) diffControlOrigination(control string,
+	openControlData opencontrols.Data) ([]reporter.Reporter, error) {
+	// find the control origination section in the document.
+	docControlOriginationData, err := newControlOrigination(st)
+	if err != nil {
+		return nil, err
+	}
+	// find the control origins currently checked in the section in the doc.
+	docControlOrigins := docControlOriginationData.getCheckedOrigins()
+
+	// find the control origins noted in the yaml.
+	yamlControlOriginationData := openControlData.GetControlOrigins(control)
+	// find the control origins currently checked in the section in the YAML.
+	yamlControlOrigins := getCheckedOriginsFromYAML(yamlControlOriginationData)
+
+	// find the difference of the two sets.
+	controlOriginMap := getControlOriginMappings()
+	reports := []reporter.Reporter{}
+
+	// find only the origins in the document.
+	onlyInDocOrigins := set.Difference(docControlOrigins, yamlControlOrigins)
+	// create the diff report for the origins only in the document.
+	onlyInDocOriginReports := createControlOriginsDiffReport(onlyInDocOrigins, controlOriginMap, control, sspSrc)
+	reports = append(reports, onlyInDocOriginReports...)
+
+	// find only the origins in the yaml.
+	onlyInYAMLOrigins := set.Difference(yamlControlOrigins, docControlOrigins)
+	// create the diff report for the origins only in the yaml.
+	onlyInYAMLOriginReports := createControlOriginsDiffReport(onlyInYAMLOrigins, controlOriginMap, control, yamlSrc)
+	reports = append(reports, onlyInYAMLOriginReports...)
+
+	return reports, nil
+}
+
+func createControlOriginsDiffReport(diff set.Interface, controlOriginMap map[controlOrigin]originMapping,
+	control string, source infoSource) []reporter.Reporter {
+	reports := []reporter.Reporter{}
+	secondField := field{text: ""}
+	for _, originInterface := range diff.List() {
+		// cast back from interface{} to controlOrigin so we can use it in the controlOriginMap
+		origin, isType := originInterface.(controlOrigin)
+		if isType {
+			var firstField field
+			switch source {
+			case sspSrc:
+				firstField.text = controlOriginMap[origin][sspSrc]
+				firstField.source = sspSrc
+				secondField.source = yamlSrc
+			case yamlSrc:
+				firstField.text = controlOriginMap[origin][yamlSrc]
+				firstField.source = yamlSrc
+				secondField.source = sspSrc
+			}
+			// Get the doc mapping and put it in the doc.
+			reports = append(reports, NewDiff(control, controlOriginationField, firstField, secondField))
+		}
```

Comment:
hmm. it shouldn't stop the loop and it's unlikely but will just print a warning out.